### PR TITLE
Only list active conda-envs for dask-gateway

### LINKED ISSUE #1159: [BUG] - Unable to create Dask clusters
### OS system and architecture in which you are running QHub

GCP

### Expected behavior

Ability to create dask clusters via dask-gateway

### Actual behavior

Unable to create Dask Clusters

### How to Reproduce the problem?

Choose the `filesystem/dask` environment OR create a new environment with the `qhub-dask` package. 

```python
from dask_gateway import Gateway
```

☝🏽 has a depreciation warning, probably not an issue right now.

```
/home/conda/filesystem/418492d4df19a27c85d04dd5e3e624b767b129c055cd3ee87f0005b5bb10318e-20220310-172659-095264-2-dask/lib/python3.9/site-packages/dask_gateway/client.py:21: FutureWarning: format_bytes is deprecated and will be removed in a future release. Please use dask.utils.format_bytes instead.
  from distributed.utils import LoopRunner, format_bytes
```

```python
gateway = Gateway()
options = gateway.cluster_options()
```

cluster_options fails with a 

```
GatewayServerError: 500 Internal Server Error

Server got itself in trouble
```

And then 

```cluster = gateway.new_cluster()```

fails with an error with a odd looking folder name

```ValueError: [Errno 2] No such file or directory: '/home/conda/admin/envs/test-admin/conda-meta'```

### Command output

```bash session
>>> options = gateway.cluster_options()

---------------------------------------------------------------------------
GatewayServerError                        Traceback (most recent call last)
Input In [4], in <cell line: 4>()
      1 from dask_gateway import Gateway
      2 gateway = Gateway()
----> 4 options = gateway.cluster_options()
      5 options

File /home/conda/dharhas@quansight.com/ef17a653925332c287418a5f465b971dedcea0bb1902e3abe411ee5c2372e0d8-20220311-013959-007907-9-dask-geo2/lib/python3.9/site-packages/dask_gateway/client.py:518, in Gateway.cluster_options(self, use_local_defaults, **kwargs)
    504 def cluster_options(self, use_local_defaults=True, **kwargs):
    505     """Get the available cluster configuration options.
    506 
    507     Parameters
   (...)
    516         A dict of cluster options.
    517     """
--> 518     return self.sync(
    519         self._cluster_options, use_local_defaults=use_local_defaults, **kwargs
    520     )

File /home/conda/dharhas@quansight.com/ef17a653925332c287418a5f465b971dedcea0bb1902e3abe411ee5c2372e0d8-20220311-013959-007907-9-dask-geo2/lib/python3.9/site-packages/dask_gateway/client.py:343, in Gateway.sync(self, func, *args, **kwargs)
    339 future = asyncio.run_coroutine_threadsafe(
    340     func(*args, **kwargs), self.loop.asyncio_loop
    341 )
    342 try:
--> 343     return future.result()
    344 except BaseException:
    345     future.cancel()

File /home/conda/dharhas@quansight.com/ef17a653925332c287418a5f465b971dedcea0bb1902e3abe411ee5c2372e0d8-20220311-013959-007907-9-dask-geo2/lib/python3.9/concurrent/futures/_base.py:446, in Future.result(self, timeout)
    444     raise CancelledError()
    445 elif self._state == FINISHED:
--> 446     return self.__get_result()
    447 else:
    448     raise TimeoutError()

File /home/conda/dharhas@quansight.com/ef17a653925332c287418a5f465b971dedcea0bb1902e3abe411ee5c2372e0d8-20220311-013959-007907-9-dask-geo2/lib/python3.9/concurrent/futures/_base.py:391, in Future.__get_result(self)
    389 if self._exception:
    390     try:
--> 391         raise self._exception
    392     finally:
    393         # Break a reference cycle with the exception in self._exception
    394         self = None

File /home/conda/dharhas@quansight.com/ef17a653925332c287418a5f465b971dedcea0bb1902e3abe411ee5c2372e0d8-20220311-013959-007907-9-dask-geo2/lib/python3.9/site-packages/dask_gateway/client.py:497, in Gateway._cluster_options(self, use_local_defaults)
    495 async def _cluster_options(self, use_local_defaults=True):
    496     url = "%s/api/v1/options" % self.address
--> 497     resp = await self._request("GET", url)
    498     data = await resp.json()
    499     options = Options._from_spec(data["cluster_options"])

File /home/conda/dharhas@quansight.com/ef17a653925332c287418a5f465b971dedcea0bb1902e3abe411ee5c2372e0d8-20220311-013959-007907-9-dask-geo2/lib/python3.9/site-packages/dask_gateway/client.py:417, in Gateway._request(self, method, url, json)
    415     raise GatewayClusterError(msg)
    416 elif resp.status == 500:
--> 417     raise GatewayServerError(msg)
    418 else:
    419     resp.raise_for_status()

GatewayServerError: 500 Internal Server Error

Server got itself in trouble


>>> cluster = gateway.new_cluster()

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Input In [3], in <cell line: 1>()
----> 1 cluster = gateway.new_cluster()

File /home/conda/filesystem/418492d4df19a27c85d04dd5e3e624b767b129c055cd3ee87f0005b5bb10318e-20220310-172659-095264-2-dask/lib/python3.9/site-packages/dask_gateway/client.py:639, in Gateway.new_cluster(self, cluster_options, shutdown_on_close, **kwargs)
    616 def new_cluster(self, cluster_options=None, shutdown_on_close=True, **kwargs):
    617     """Submit a new cluster to the gateway, and wait for it to be started.
    618 
    619     Same as calling ``submit`` and ``connect`` in one go.
   (...)
    637     cluster : GatewayCluster
    638     """
--> 639     return GatewayCluster(
    640         address=self.address,
    641         proxy_address=self.proxy_address,
    642         public_address=self._public_address,
    643         auth=self.auth,
    644         asynchronous=self.asynchronous,
    645         loop=self.loop,
    646         cluster_options=cluster_options,
    647         shutdown_on_close=shutdown_on_close,
    648         **kwargs,
    649     )

File /home/conda/filesystem/418492d4df19a27c85d04dd5e3e624b767b129c055cd3ee87f0005b5bb10318e-20220310-172659-095264-2-dask/lib/python3.9/site-packages/dask_gateway/client.py:814, in GatewayCluster.__init__(self, address, proxy_address, public_address, auth, cluster_options, shutdown_on_close, asynchronous, loop, **kwargs)
    802 def __init__(
    803     self,
    804     address=None,
   (...)
    812     **kwargs,
    813 ):
--> 814     self._init_internal(
    815         address=address,
    816         proxy_address=proxy_address,
    817         public_address=public_address,
    818         auth=auth,
    819         cluster_options=cluster_options,
    820         cluster_kwargs=kwargs,
    821         shutdown_on_close=shutdown_on_close,
    822         asynchronous=asynchronous,
    823         loop=loop,
    824     )

File /home/conda/filesystem/418492d4df19a27c85d04dd5e3e624b767b129c055cd3ee87f0005b5bb10318e-20220310-172659-095264-2-dask/lib/python3.9/site-packages/dask_gateway/client.py:919, in GatewayCluster._init_internal(self, address, proxy_address, public_address, auth, cluster_options, cluster_kwargs, shutdown_on_close, asynchronous, loop, name)
    917     self.status = "starting"
    918 if not self.asynchronous:
--> 919     self.gateway.sync(self._start_internal)

File /home/conda/filesystem/418492d4df19a27c85d04dd5e3e624b767b129c055cd3ee87f0005b5bb10318e-20220310-172659-095264-2-dask/lib/python3.9/site-packages/dask_gateway/client.py:343, in Gateway.sync(self, func, *args, **kwargs)
    339 future = asyncio.run_coroutine_threadsafe(
    340     func(*args, **kwargs), self.loop.asyncio_loop
    341 )
    342 try:
--> 343     return future.result()
    344 except BaseException:
    345     future.cancel()

File /home/conda/filesystem/418492d4df19a27c85d04dd5e3e624b767b129c055cd3ee87f0005b5bb10318e-20220310-172659-095264-2-dask/lib/python3.9/concurrent/futures/_base.py:446, in Future.result(self, timeout)
    444     raise CancelledError()
    445 elif self._state == FINISHED:
--> 446     return self.__get_result()
    447 else:
    448     raise TimeoutError()

File /home/conda/filesystem/418492d4df19a27c85d04dd5e3e624b767b129c055cd3ee87f0005b5bb10318e-20220310-172659-095264-2-dask/lib/python3.9/concurrent/futures/_base.py:391, in Future.__get_result(self)
    389 if self._exception:
    390     try:
--> 391         raise self._exception
    392     finally:
    393         # Break a reference cycle with the exception in self._exception
    394         self = None

File /home/conda/filesystem/418492d4df19a27c85d04dd5e3e624b767b129c055cd3ee87f0005b5bb10318e-20220310-172659-095264-2-dask/lib/python3.9/site-packages/dask_gateway/client.py:933, in GatewayCluster._start_internal(self)
    931     self._start_task = asyncio.ensure_future(self._start_async())
    932 try:
--> 933     await self._start_task
    934 except BaseException:
    935     # On exception, cleanup
    936     await self._stop_internal()

File /home/conda/filesystem/418492d4df19a27c85d04dd5e3e624b767b129c055cd3ee87f0005b5bb10318e-20220310-172659-095264-2-dask/lib/python3.9/site-packages/dask_gateway/client.py:946, in GatewayCluster._start_async(self)
    944 if self.status == "created":
    945     self.status = "starting"
--> 946     self.name = await self.gateway._submit(
    947         cluster_options=self._cluster_options, **self._cluster_kwargs
    948     )
    949 # Connect to cluster
    950 try:

File /home/conda/filesystem/418492d4df19a27c85d04dd5e3e624b767b129c055cd3ee87f0005b5bb10318e-20220310-172659-095264-2-dask/lib/python3.9/site-packages/dask_gateway/client.py:535, in Gateway._submit(self, cluster_options, **kwargs)
    533     options = self._config_cluster_options()
    534     options.update(kwargs)
--> 535 resp = await self._request("POST", url, json={"cluster_options": options})
    536 data = await resp.json()
    537 return data["name"]

File /home/conda/filesystem/418492d4df19a27c85d04dd5e3e624b767b129c055cd3ee87f0005b5bb10318e-20220310-172659-095264-2-dask/lib/python3.9/site-packages/dask_gateway/client.py:413, in Gateway._request(self, method, url, json)
    410     msg = await resp.text()
    412 if resp.status in {404, 422}:
--> 413     raise ValueError(msg)
    414 elif resp.status == 409:
    415     raise GatewayClusterError(msg)

ValueError: [Errno 2] No such file or directory: '/home/conda/admin/envs/test-admin/conda-meta'
```

### Versions and dependencies used.

conda 4.11.0
QHub 0.4 RC

### Compute environment

GCP

### Integrations

Dask

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by iameskild**
Fixes | Closes | Resolves #1159

> Please remove anything marked as optional that you don't need to fill in.
> Choose one of the keywords preceding to refer to the issue this PR solves, followed by the issue number (e.g Fixes # 666).
> If there is no issue, remove the line. Remove this note after reading.

## Changes introduced in this PR:

- Bug fix for issue 1159

## Types of changes

What types of changes does your PR introduce?

_Put an `x` in the boxes that apply_

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

### Requires testing

- [ ] Yes
- [ ] No

### In case you checked yes, did you write tests?

- [ ] Yes
- [ ] No

## Documentation

Does your contribution include breaking changes or deprecations?
If so have you updated the documentation?

- [ ] Yes, docstrings
- [ ] Yes, main documentation
- [ ] Yes, deprecation notices

## Further comments (optional)

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered and more.

### CODE REVIEW

**Comment by iameskild** [comment]
I made a minor change to how the Github workflow files are rendered. Namely, added `QHUB_GH_BRANCH = {{ secret.QHUB_GH_BRANCH }}` as an environment variable. This will allow users to deploy qhub from a branch using CI/CD by simply adding `QHUB_GH_BRANCH` as a secret. 

I decided to add it to this PR since I wanted to test the first fix and this subsequent change on `qhub-deployment-quansight-beta`. This method was tested and was successful :)

https://github.com/Quansight/qhub-deployment-quansight-beta/actions/runs/1986189931

File: qhub/template/stages/07-kubernetes-services/modules/kubernetes/services/dask-gateway/files/gateway_config.py [hunk 1]
```diff
@@ -86,12 +86,18 @@ async def authenticate(self, request):
...
+        for filename in os.listdir(os.path.join(conda_prefix, "conda-meta")):
+            if filename.endswith(".json"):
+                with open(os.path.join(conda_prefix, "conda-meta", filename)) as f:
+                    packages.add(json.load(f).get("name"))
+        return packages
+    except OSError as e:
+        import logging
+
+        logger = logging.getLogger()
+        logger.warning(f"An issue with a conda environment was encountered.\n{e}")
```
[the remark this answers is not in the document]
Comment (iameskild) [reply qhub/template/stages/07-kubernetes-services/modules/kubernetes/services/dask-gateway/files/gateway_config.py]: Sounds good to me!
Done.

**Comment by iameskild** [comment]
By specifying the `jupyter_server` versions which are known to work, we can get around the issue outlined in #1166.

**Comment by iameskild** [comment]
The latest `jupyter_server` version resolves the issues we were experiencing so it has been removed as a dependency.

**Comment by viniciusdc** [comment]
> The latest `jupyter_server` version resolves the issues we were experiencing so it has been removed as a dependency.

Thanks Eskild for checking this. Though as we dont request this dependecy directly, could you add `jupyter_server ~= < problematic version >`

**Comment by iameskild** [comment]
That's a good call @viniciusdc!

**Comment by costrouc** [comment]
Rerunning the tests just so that we get a pass on the tests

**Review by viniciusdc — APPROVED** [review]
LGTM! Thanks, Eskild.

**Pull request merged**